media/server/gstplayer/source/tasks/webAudio/SetCaps.cpp

```diff
@@ -102,14 +102,14 @@ class WebAudioPcmCapsBuilder : public WebAudioCapsBuilder
 
 SetCaps::SetCaps(WebAudioPlayerContext &context, std::shared_ptr<IGstWrapper> gstWrapper,
                  std::shared_ptr<IGlibWrapper> glibWrapper, const std::string &audioMimeType,
-                 const WebAudioConfig *config)
+                 std::weak_ptr<const WebAudioConfig> config)
     : m_context{context}, m_gstWrapper{gstWrapper}, m_glibWrapper{glibWrapper}, m_audioMimeType{audioMimeType}
 {
     RIALTO_SERVER_LOG_DEBUG("Constructing SetCaps");
 
     if (m_audioMimeType == "audio/x-raw")
     {
-        m_config.pcm = config->pcm;
+        m_config.pcm = config.lock()->pcm;
```

Comment:
Using lock without nullptr check may be dangerous :)

Reply:
@skywojciechowskim the ptr can be null therefore I added a check that throws an appropriate error before using (similar to media/client/main/source/WebAudioPlayer.cpp:92 and `media/server/main/source/WebAudioPlayerServerInternal.cpp:105)